parlai/core/torch_classifier_agent.py

```diff
@@ -158,25 +161,26 @@ def __init__(self, opt: Opt, shared=None):
             if init_model:
                 print('Loading existing model parameters from ' + init_model)
                 self.load(init_model)
-            if self.use_cuda:
-                if self.opt['data_parallel']:
-                    if is_distributed():
-                        raise ValueError(
-                            'Cannot combine --data-parallel and distributed mode'
-                        )
-                    self.model = torch.nn.DataParallel(self.model)
+        if self.use_cuda:
+            if self.opt['data_parallel']:
+                if is_distributed():
+                    raise ValueError(
+                        'Cannot combine --data-parallel and distributed mode'
+                    )
+                self.model_data_parallel = torch.nn.DataParallel(self.model)
+                self.model = self.model_data_parallel.module
```

Comment:
does this not mess up later computation? I think we should be calling `self.model()` rather than `self.model_data_parallel()`

Reply:
yeah i agree. what is going on here? i am pretty sure it worked fine before